specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/preview/2024-08-22-preview/confidentialledger.json

```diff
@@ -0,0 +1,1862 @@
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "CreateOrUpdateLedgerUsers",
+        "x-ms-examples": {
+          "CreateOrUpdateLedgerUsers": {
+            "$ref": "./examples/ledger/CreateOrUpdateLedgerUsers.json"
+          }
+        },
+        "summary": "Adds a user or updates a user's fields.",
+        "description": "A JSON merge patch is applied for existing users",
+        "consumes": [
+          "application/merge-patch+json"
+        ],
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/UserIdParameter"
+          },
+          {
+            "$ref": "#/parameters/UserMultipleRolesParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "User created or updated successfully.",
+            "schema": {
+              "$ref": "#/definitions/LedgerUserMultipleRoles"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/userDefinedEndpoints": {
+      "get": {
+        "operationId": "GetUserDefinedEndpoint",
+        "x-ms-examples": {
+          "GetUserDefinedEndpoint": {
+            "$ref": "./examples/ledger/GetUserDefinedEndpoint.json"
+          }
+        },
+        "summary": "Gets a user defined endpoint.",
+        "description": "Returns the user defined endpoint in the ACL instance",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Gets the user defined endpoints.",
+            "schema": {
+              "$ref": "#/definitions/Bundle"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      },
+      "put": {
+        "operationId": "CreateUserDefinedEndpoint",
+        "x-ms-examples": {
+          "CreateUserDefinedEndpoint": {
+            "$ref": "./examples/ledger/CreateUserDefinedEndpoint.json"
+          }
+        },
+        "summary": "Creates a user defined endpoint.",
+        "description": "Creates the user defined endpoint in the ACL instance ",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/BundleParameter"
+          }
```

Comment:
You should add explicit support for `If-Match` header with etag to allow users to reliably update/replace endpoints.

Reply:
Sure, will add this in the upcoming GA version